mesonbuild/coredata.py

```diff
@@ -335,11 +335,10 @@ def sanitize_dir_option_value(self, prefix, option, value):
     def init_builtins(self):
         # Create builtin options with default values
         self.builtins = {}
-        prefix = get_builtin_option_default('prefix')
-        for key in get_builtin_options():
-            value = get_builtin_option_default(key, prefix)
-            args = [key] + builtin_options[key][1:-1] + [value]
-            self.builtins[key] = builtin_options[key][0](*args)
+        for key, opt in builtin_options.items():
```

Comment:
nitpick: `builtin_options` should be declared `global builtin_options` in each function that uses it (yes there are pre-existing cases below, but I think those should be changed too)

Reply:
no, those absolutely should not use the global keyword. `global` is only for assigning variables from the global scope inside a scope:
```python
foo = None

def setfoo(new):
   global foo = new

setfoo()
```
It really is just for avoiding shadowing. As a python guy, using global is *always* a sign that you have a misdesign

Reply:
Is it considered ok according to python convention to access global variables without explicitly listing them as such?

```python
foo = {'1': 'a', '2': 'b'}

def func(foo):
    global foo
    for each in foo.items():
        pass
```

This won't assign anything to `foo` or change it but I feel like it makes it clearer that `foo` is being read from the global scope.

Reply:
That would be very strange. Mutating globals is certainly a little smelly, but that's python's "we're all consenting adults", `global` really is just for assigning at the global level.

I'm assuming that you didn't mean to pass `foo` as an argument and `global` it?

Reply:
> I'm assuming that you didn't mean to pass `foo` as an argument and `global` it?

Yes, that's a typo.